ast/misc.go

```diff
@@ -414,6 +414,20 @@ type UserSpec struct {
 	AuthOpt *AuthOption
 }
 
+// SecurityString formats the UserSpec without password information.
+func (u *UserSpec) SecurityString() string {
+	withPassword := false
+	if opt := u.AuthOpt; opt != nil {
+		if len(opt.AuthString) > 0 || len(opt.HashString) > 0 {
+			withPassword = true
+		}
+	}
+	if withPassword {
+		return fmt.Sprintf("{%s password = ***}", u.User)
```

Comment:
It doesn't matter as long as the log doesn't contain password.
I choose to distinguish empty password because it maybe helpful for debugging.